data_analysis/schedule_manager.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+from dataclasses import dataclass
+import zipfile
+from tqdm import tqdm
+from typing import List, Tuple
+
+import pendulum
+import logging
+import pandas as pd
+
+from scrape_data.scrape_schedule_versions import create_schedule_list
+from data_analysis.gtfs_fetcher import GTFSFetcher, ScheduleFeedInfo, ScheduleSource
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO)
+logger.setLevel(logging.INFO)
+
+BASE_URL = "https://transitfeeds.com"
+
+# Last historical schedule available on transitfeeds.com
+LAST_TRANSITFEEDS = pendulum.date(2023, 12, 7)
```

Comment:
That would be good to investigate. In this PR I wanted to not change the previous data sources if possible, so that's why I left with that date. Could we investigate switching in a future PR? Note that as above this cutover date is only relevant when rebuilding the entire history.

Reply:
Split to new issue: https://github.com/chihacknight/chn-ghost-buses/issues/85